test/unit/lib/plugins/aws/package/compile/events/kafka.test.js

```diff
@@ -0,0 +1,534 @@
+          },
+        },
+        SourceAccessConfigurations: [
+          {
+            Type: 'SASL_SCRAM_256_AUTH',
+            URI: saslScram256AuthArn,
+          },
+        ],
+        StartingPosition: 'TRIM_HORIZON',
+        Topics: [topic],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('foo'), 'Arn'],
+        },
+      });
+    });
+
+    it('should update default IAM role with SecretsManager statement', () => {
+      expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).to.deep.include({
+        Effect: 'Allow',
+        Action: ['secretsmanager:GetSecretValue'],
+        Resource: [saslScram256AuthArn],
+      });
+    });
+
+    it('should correctly compile EventSourceMapping resource DependsOn ', () => {
+      expect(minimalEventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+      expect(allParamsEventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+    });
+
+    it('should correctly compile EventSourceMapping resource with all parameters', () => {
+      expect(allParamsEventSourceMappingResource.Properties).to.deep.equal({
+        BatchSize: batchSize,
+        Enabled: enabled,
+        SelfManagedEventSource: {
+          Endpoints: {
+            KafkaBootstrapServers: ['abc.xyz:9092'],
+          },
+        },
+        SourceAccessConfigurations: [
+          {
+            Type: 'SASL_SCRAM_256_AUTH',
+            URI: saslScram256AuthArn,
+          },
+        ],
+        StartingPosition: startingPosition,
+        Topics: [topic],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('other'), 'Arn'],
+        },
+      });
+    });
+  });
+
+  describe('configuring kafka events', () => {
+    const runCompileEventSourceMappingTest = async (eventConfig) => {
+      const { awsNaming, cfTemplate } = await runServerless({
+        fixture: 'function',
+        configExt: {
+          functions: {
+            foo: {
+              events: [
+                {
+                  kafka: eventConfig.event,
+                },
+              ],
+            },
+          },
+        },
+        cliArgs: ['package'],
+      });
+
+      const eventSourceMappingResource =
+        cfTemplate.Resources[awsNaming.getKafkaEventLogicalId('foo', 'TestingTopic')];
+      expect(eventSourceMappingResource.Properties).to.deep.equal(eventConfig.resource(awsNaming));
+    };
+
+    const requirePropertyTest = async (event) => {
+      await expect(
+        runServerless({
+          fixture: 'function',
+          configExt: {
+            functions: {
+              foo: {
+                events: [
+                  {
+                    kafka: event,
+                  },
+                ],
+              },
+            },
+          },
+          cliArgs: ['package'],
+        })
+      ).to.be.eventually.rejectedWith(ServerlessError);
+    };
+
+    it('should require access configurations', async () => {
```

Comment:
As @pgrzesik pointed, let's not introduce tests that confirm that schema validation errors are thrown on invalid values

We have some sanity tests for it in other place, and otherwise schema validation engine is extensively tested in its own repository.